### ISSUE TITLE
FPS drop with Reshade and 4K set in ini

### ISSUE THREAD

**Issue by deton24** [AeroWidescreen/NFSMWHDReflections#1]
FPS drop with Reshade and 4K set in ini and/or with Vsync.
From 60-70 to 40 when Reshade with some preset is enabled.
When I disable Reshade, it backs to normal.
Leaving default FHD or 16:10 in mod's ini, fixes the framerate to ~52.
Yes, maybe it's because of resources.
Additionally, enabling Vsync causes even greater FPS drops. If Reshade is disabled, they disapears too.
AMD RX Series.

**Comment by AeroWidescreen** [comment]
> Yes, maybe it's because of resources.

And the game is old and has performance issues. For example, at 4K resolution my framerate stays at around 45fps in the garage until I disable vsync. That's _without_ HD Reflections installed, so I know that's not the problem.

With everything working together: 4K resolution, 4K textures, 4K shadows, 4K reflections, etc., I get anywhere between 50% to 90% GPU usage during gameplay, and it gets worse if ImproveReflectionLOD is enabled. Now add ReShade to the mix and I wouldn't be surprised if your GPU was bottlenecking.

**Comment by AeroWidescreen** [comment]
> Yes, maybe it's because of resources.

And the game is old and has issues. For example, at 4K resolution my framerate stays at around 45fps in the garage until I disable vsync. That's _without_ the HD Reflection mod installed. In contrast, staying at 1080p with 4K reflections doesn't give me any trouble at all.

With everything working together: 4K resolution, 4K textures, 4K shadows, 4K reflections, etc., I get anywhere between 50% to 90% GPU usage during gameplay, and it gets worse if ImproveReflectionLOD is enabled. 

Now consider that you're using ReShade with all of this and I wouldn't be surprised if your GPU was bottlenecking.

In other words, you have to lower your display resolution. There's nothing I can do about it since this isn't an issue caused by HD Reflections as far as I'm aware.

**Comment by deton24** [comment]
Thanks for your response.

I just finished up with lowering shadows resolution, and it is still not bad, because I don't really have to lower my game resolution.

Probably not much below FHD for shadow resolution will be enough to keep 60 FPS. On FHD setting I have some small drops here and there.
Anyway, GPU usage may not be the relevant point of view. Probably more powerful GPU would handle it.
Looks like overclocked RX 470 is not really enough. But still, close.

**Comment by AeroWidescreen** [comment]
> Thanks for your response.

No problem, thanks for your report!

> Anyway, GPU usage may not be the relevant point of view. Probably more powerful GPU would handle it.

I have an 8GB RX 580 and the results aren't  much better. You'd probably need something like a GTX 1070 or RTX 2060.

The garage running at 45fps is only a vsync bug, but during gameplay I still drop below 60fps when there's a lot of cars + rain. Look at my GPU usage here:
![bottleneck](https://i.imgur.com/ErFZYMp.jpg)

I'm glad you were able to get better results by lowering the shadow resolution though. Shadows, anti-aliasing, and resolution are always the most resource heavy features from my experience.
